### DISCUSSION TITLE
Feature Request: Indentation Block

### DISCUSSION THREAD

**Discussion by kevintraver** [chrisgrieser/nvim-various-textobjs#43] [Ideas]
Here's an idea for a new text-object that might be useful: _Indentation Block_ (open to ideas on other names this could be called)

Essentially, its the same as the indentation text object, but restricted by newlines.

example

```
myFunction (abc)
  callMethod

  doSomething
  doSomethingElse
  doAnotherThing

  anotherMethod
end
```

if the cursor was on **_doSomethingElse_**, it would select the following
```
  doSomething
  doSomethingElse
  doAnotherThing

```

**Comment by chrisgrieser** [comment]
wouldn't that be the same as the paragraph textobject `ip`?

**Reply by bew** [reply]
Not in this case:
```
myFunction (abc)
  callMethod
  doSomething  // cursor here
  doSomethingElse

  doAnotherThing
  anotherMethod
end
```

Where the wish would be to select:
```
  callMethod
  doSomething
  doSomethingElse
```

It has a vimscript implementation already in https://github.com/kana/vim-textobj-indent , under `ii` omap. (`ai` omap of this plugin is the whole content of the function, everything with same indentation as current line)

**Reply by chrisgrieser** [reply]
I see, so a combination of indentation and paragraph textobject, essentially.

Could you file FR-issue? Then I'll see to it when I got a bit more time at hand.

**Reply by kevintraver** [reply]
Yes, that's correct. It would be a different object than paragraph.

As shown above,, this would be a lot more useful than `ip` when editing lots of code.

**Reply by kevintraver** [reply]
I've created a draft PR #54 that would potentially work.

**Comment by chrisgrieser** [comment]
closing since implemented a while ago
